packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -496,7 +496,27 @@ class VotesRepo extends AbstractRepo<"Votes"> {
     );
     return results.map(({ vote_id }) => vote_id);
   }
-
+  
+  async getVotesOnSamePost({userId, postId, excludedDocumentId}: {
+    userId: string,
+    postId: string,
+    excludedDocumentId: string,
+  }): Promise<DbVote[]> {
+    return await this.manyOrNone(`
+      SELECT
+        v.*
+      FROM
+        "Votes" v,
+        "Comments" c
+      WHERE
+        v."cancelled" IS FALSE
+        AND v."userId" = $1
+        AND v."documentId" != $3
```

Comment:
This is to exclude the user's existing vote (if any) on the specific comment which they're currently voting on, which triggered the check for this particular rate limit?

Reply:
Right, since any old vote on that comment is in the process of being replaced by a new vote.